mono_repo/lib/src/commands/list_command.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+import '../package_config.dart';
+import '../root_config.dart';
+import 'mono_repo_command.dart';
+
+class ListCommand extends MonoRepoCommand {
+  ListCommand() {
+    argParser
+      ..addFlag(
+        'only-published',
+        abbr: 'p',
+        help: 'Only list packages with a version and without publish_to set to '
+            'none.',
+      )
+      ..addMultiOption(
+        'show',
+        abbr: 's',
+        help:
+            'The properties of the package to show in a comma-seperated list.',
+        allowed: ShowItem.values.map((e) => e.name),
+        allowedHelp: Map.fromEntries(
+          ShowItem.values.map((e) => MapEntry(e.name, e.help)),
+        ),
+        defaultsTo: ShowItem.values
+            .where((element) => element.defaultsTo)
+            .map((e) => e.name),
+      );
+  }
+
+  @override
+  String get name => 'list';
+
+  @override
+  String get description => 'List all packages configured for mono_repo.';
+
+  @override
+  void run() => print(listPackages(
+        rootConfig(),
+        onlyPublished: argResults!['only-published'] as bool,
+        showItems: (argResults!['show'] as List<String>)
+            .map((e) =>
+                ShowItem.values.singleWhere((element) => element.name == e))
+            .toSet(),
+      ).join('\n'));
+}
+
+enum ShowItem {
```

Comment:
This feel very generic - the context we have is "list_command", and this is "showItem". The items and the list could be anything.

How about changing the library to `list_packages_command`? After that it's not a big deal if this name is very generic since the context will be less generic. I'd still consider changing it though, I typically would associate "Item" in the context of "list" to be one of the rows, now one of the columns. Personally I'd use either `Field` or maybe `PackageField` here

Reply:
Or maybe even `column`.

Reply:
went with column!